nebullvm/operations/optimizations/compilers/openvino.py

```diff
@@ -25,6 +27,130 @@
 from nebullvm.tools.transformations import MultiStageTransformation
 
 
+
+class TensorFlowOpenVINOCompiler(Compiler):
+    supported_ops = {
+        "cpu": [
+            None,
+            # QuantizationType.STATIC,
+            # QuantizationType.HALF,
+        ],
+        "gpu": [],
+    }
+
+    def __init__(self):
+        super().__init__()
+
+    def execute(
+        self,
+        model: tf.keras.Model,
+        model_params: ModelParams,
+        input_tfms: MultiStageTransformation = None,
+        metric_drop_ths: float = None,
+        quantization_type: QuantizationType = None,
+        input_data: DataManager = None,
+        **kwargs,
+    ):
+        """Compile the input model using TF-OPENVINO library.
+
+        Args:
+            model (tf.keras.Model()): The tensorflow model.
+            model_params (ModelParams): The model parameters.
+            output_file_path (str or Path): Path where storing the output
+            TensorFlow file.
+            input_tfms (MultiStageTransformation, optional): Transformations
+                to be performed to the model's input tensors in order to
+                get the prediction. Default: None.
+            metric_drop_ths (float, optional): Threshold for the accepted drop
+                in terms of precision. Any optimized model with a higher drop
+                will be ignored. Default: None.
+            quantization_type (QuantizationType, optional): The desired
+                quantization algorithm to be used. Default: None.
+            input_data (DataManager): User defined data. Default: None
+        """
+
+        if quantization_type not in self.supported_ops[self.device.value]:
+            self.compiled_model = None
+            return
+
+        if quantization_type is QuantizationType.STATIC and input_data is None:
+            raise ValueError("Input data is required for static quantization.")
+
+        self.logger.info(
+            f"Optimizing with {self.__class__.__name__} and "
+            f"q_type: {quantization_type}."
+        )
+
+        check_quantization(quantization_type, metric_drop_ths)
+        train_input_data = input_data.get_split("train").get_numpy_list(
+            QUANTIZATION_DATA_NUM
+        )
+
+
+        # save model to temporary folder   
+        tmp_dir_path = Compiler.onnx_output_path     
```

Comment:
Be careful, the Compiler class doesn't have an `onnx_output_path` attribute. 
`onnx_output_path` is an argument passed to this method from the caller (it's passed to each execute method of each compiler), so you should add it to the arguments above of the execute (line 46->51) like this:
```python
def execute(
     self,
     model: tf.keras.Model,
     model_params: ModelParams,
     onnx_output_path: str,
     input_tfms: MultiStageTransformation = None,
     metric_drop_ths: float = None,
     quantization_type: QuantizationType = None,
     input_data: DataManager = None,
     **kwargs,
 ):
```

This parameters was introduced in the `DeepSparseCompiler`, that needs to export the original PyTorch model to onnx and so it needs the path to a temporary dir to generate the onnx model (that's why it's called onnx_output_path). In this case we are going to use it to convert a model to openvino, so I would suggest to rename it to `tmp_dir_path` that is a more general name (please reame it also in the DeepSparseCompiler and in the caller).